src/sentry/api/endpoints/sentry_internal_app_token_details.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+from rest_framework import status
+from django.http import Http404
+
+from sentry.api.bases import (
+    SentryInternalAppTokenPermission, SentryAppBaseEndpoint,
+)
+from sentry.models import ApiToken
+from sentry.features.helpers import requires_feature
+from sentry.mediators.sentry_app_installation_tokens import Destroyer
+
+
+class SentryInternalAppTokenDetailsEndpoint(SentryAppBaseEndpoint):
+    permission_classes = (SentryInternalAppTokenPermission, )
+
+    def convert_args(self, request, sentry_app_slug, token, *args, **kwargs):
+        # get the sentry_app from the SentryInternalAppTokenDetailsEndpoint class
+        (args, kwargs) = super(SentryInternalAppTokenDetailsEndpoint,
+                               self).convert_args(request, sentry_app_slug, *args, **kwargs)
+
+        try:
+            kwargs['api_token'] = ApiToken.objects.get(token=token)
+        except ApiToken.DoesNotExist:
+            raise Http404
+
+        return (args, kwargs)
+
+    @requires_feature('organizations:sentry-apps', any_org=True)
+    def delete(self, request, sentry_app, api_token):
+        if not sentry_app.is_internal:
+            return Response('This route is limited to internal integrations only',
+                            status=status.HTTP_403_FORBIDDEN
```

Comment:
This error message exposes that both the sentry app and token both exist (potentially on another organization). Are we ok disclosing that information to a potential attacker?

Reply:
@markstory Unless I am mistaken, I don't think that is true. The `convert_args` method of `SentryAppBaseEndpoint` which we are using in the `super` call of this class will check the permissions of the `sentry_app` before we get there. So you won't get to `403` error unless you have the permissions for that `sentry_app`. And as for permission issues with the token, as mentioned in my other comment we check that the API token matches the sentry app and we throw a `404` if they don't so an attacker would not know the token exists.